src/api/File.js

```diff
@@ -78,29 +79,71 @@ class File extends Item {
         this.successHandler(`${downloadBaseUrl}?${downloadUrlQuery}`);
     }
 
+    /**
+     * Polls an item's infoUrl until it's representation is generated or an error occurrs.
+     * Calls callback if representation was successfully generated.
+     *
+     * @param {BoxItem} item - BoxItem that needs infoUrl polled
+     * @param {string} thumbanilUrl - thumbanil url of item
+     * @param {string, string) => void} callback - function to call if polling results in success
+     * @return {Promise<void>}
+     */
+    async pollInfoUrl(
+        item: BoxItem,
+        thumbnailUrl: string,
+        callback: (id: string, thumbnailUrl: string) => void = noop,
+    ): Promise<void> {
+        const { id } = item;
+        const infoUrl = getProp(item, 'representations.entries[0].info.url');
+
+        if (!infoUrl) {
+            return;
+        }
+
+        let status;
+        do {
+            // eslint-disable-next-line no-await-in-loop
```

Comment:
From [https://eslint.org/docs/rules/no-await-in-loop](url) on when not to use `no-await-in-loop` rule:
>loops may be used to retry asynchronous operations that were unsuccessful

I think this scenario fits that case.